src/cpp/fastdds/subscriber/DataReaderImpl.cpp

```diff
@@ -238,6 +240,13 @@ ReturnCode_t DataReaderImpl::enable()
         return ReturnCode_t::RETCODE_ERROR;
     }
 
+    auto topic_impl = topic_->get_impl();
```

Comment:
You're getting the `topic_impl` twice (the other being the setting the property). Moving this line up we can do it just once